### PR TITLE
staking & unstaking

### PR DESCRIPTION
**Pull request by rindeveloper**

### CODE REVIEW

File: src/api/apiPoolsClient.ts [hunk 1]
```diff
@@ -103,8 +101,36 @@ export class AldrinApiPoolsClient extends AldrinApiClient {
...
+            farmingSnapshots
+            tokensUnlocked
+            tokensTotal
+            tokensPerPeriod
+            periodLength
+            vestingPeriod
+            currentTime
+          }
+        }
+      }
```
Comment (jaburx) [inline src/api/apiPoolsClient.ts:129]: should add variables so we fetch a specific staking pool.. not a random.

File: src/api/apiPoolsClient.ts [hunk 2]
```diff
@@ -103,8 +101,36 @@ export class AldrinApiPoolsClient extends AldrinApiClient {
...
+    `
+
+    const response = await this.gqlClient.request<GetStakingPoolInfoResponse>(query)
 
-    return parsed
+    return response.getStakingPoolInfo
```
Comment (jaburx) [inline src/api/apiPoolsClient.ts:134]: add Apollo or Relay for caching, garbage collector... etc.. free powers.

File: src/api/types.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import { PublicKey } from '@solana/web3.js';
 import BN from 'bn.js';
-import { FarmingState, PoolCommon, WithPoolPK } from '..';
+import { PoolCommon, WithPoolPK } from '..';
```
Comment (jaburx) [inline src/api/types.ts:3]: try to avoid abbreviations. WithPoolPK might be privateKey

File: src/farming/client.ts [hunk 4]
```diff
@@ -227,7 +226,7 @@ export class FarmingClient {
     )
 
     return sendTransaction({
-      wallet: wallet,
+      wallet,
```
Comment (jaburx) [inline src/farming/client.ts:229]: nice

**Review by jaburx — APPROVED** [review]
looks good. I'd add more tests... 50% of the sdk fns has something broken works on the SDK.

**Review by 0xrinegade — APPROVED** [review]
LGTM

**Pull request merged**